include/pagmo/algorithms/pso.hpp

```diff
@@ -254,9 +271,10 @@ class PAGMO_DLL_PUBLIC pso
     unsigned m_neighb_param;
     // memory
     bool m_memory;
-    // paricles' velocities
-    mutable std::vector<vector_double> m_V;
+    mutable boost::optional<pso::memory> m_memory_data;
 
+    // particles' velocities
+    mutable std::vector<vector_double> m_V;
```

Comment:
I wonder if I should move this to `memory`, as it is the only variable that now has two semantics. (1) the usual `m_V` used as an variable in the algorithm (the others are local variables) and (2) as a memory variable.

Let me know if this makes sense to you.